controllers/competition_supervisor/competition_supervisor.py

```diff
@@ -193,6 +193,15 @@ def get_simulation_run_mode(supervisor: Supervisor) -> SimulationMode:
         return Supervisor.SIMULATION_MODE_FAST
 
 
+def set_simulation_mode(supervisor: Supervisor, mode: SimulationMode) -> None:
+    try:
+        supervisor.simulationSetMode(mode)
+    except AttributeError:
+        # Webots R2023a throws an AttributeError so we'll call underlying setter ourselves
```

Comment:
This makes it sound like a bug -- is that expected? If so, should we report this upstream?

Separately I'd be tempted to guard _only_ the attribute lookup we're using as detection of the runtime version, so that any other `AttributeError`s don't cascade into hard-to-diagnose issues.

Reply:
This bug has actually been fixed upstream and will be in the next release. https://github.com/cyberbotics/webots/blob/master/docs/reference/changelog-r2023.md?plain=1#L15